### ISSUE TITLE
Cannot render form filter

### ISSUE THREAD

**Issue by menonsoftware** [elapouya/django-listing#18]
Hello,

I'm using Django-listing to display data which should be filtered using form filters. I can list the data, however I'm getting an error displaying the form stating: "
**ERROR :** The listing must be initialised before rendering the filter form. You should use {% create_listing a_listing_class your_data as listing %} tag before in the template or provide in the context a listing instance bound to some data."

I'm using Python side rendering and am able to filter data using the URL (http://localhost:8000/index?f_transaction_id=00000), just that the form does not get registered in the UI. Where am I going wrong?

The template is as below:
==============================================================

```
          <div class="accordion accordion-flush" id="accordion">
            <div class="accordion-item">
              <h2 class="accordion-header">
                <button class="accordion-button collapsed"
                        type="button"
                        data-bs-toggle="collapse"
                        data-bs-target="#flush-collapseOne"
                        aria-expanded="false"
                        aria-controls="flush-collapseOne">Tracker Filters</button>
              </h2>
              <div id="flush-collapseOne"
                   class="accordion-collapse collapse"
                   data-bs-parent="#accordion">                 
                    {% render_filters_form tracker %}
                </div>
              </div>
            </div>
            <br />
            {% render_listing tracker %}
          </div>
```

==============================================================

Views.py

```
class index(ListingView):
    template_name = 'trackers/index.html'

    context_classes = (
        ToolbarListing,
        Tracker,
    )
         
    def get_context_data(self, **kwargs):
        context = super(index, self).get_context_data(**kwargs)
        if is_manager_admin(self.request.user):
            context["tracker"] = ToolbarListing(Tracker.objects.all())
        else:
            context["tracker"] = ToolbarListing(Tracker.objects.filter(email=self.request.user.email))
        return context

```

listings.py

```
class ToolbarListing(ListingVariations):   
    variations_classes = (
        ToolbarSimpleListing,
        ToolbarEmployeeDivListing,
        ToolbarEmployeeThumbnailsListing,
        # ToolbarEmployeeBigThumbnailsListing,
    )
    
    toolbar = Toolbar(
        ExportSelectToolbarItem(),
        SortSelectToolbarItem(),
        VariationsToolbarItem(
            labels=("Listing", "Detailed", "Thumbnails", "Big thumbnails"),
            icons=(
                "listing-icon-menu-2",
                "listing-icon-th-list-4",
                "listing-icon-th-3",
                # "listing-icon-th-large-2",
            ),
        ),
        PerPageSelectToolbarItem(choices="3,8,16,32,64,-1:All"),
    )
    toolbar_placement = "both"
    columns = Columns( Column('colleague'), Column('transaction_id'), Column('family_name'), Column('record_date'),Column('record_time') )
    paginator_has_first_last = True
    paginator_page_scale_size = 9
    paginator_fast_page_prev_tpl = "Previous {step} pages"
    paginator_fast_page_next_tpl = "Next {step} pages"
    paginator_fast_page_step = 10
    paginator_theme_fast_page_has_icon = False
    paginator_theme_next_icon = "listing-icon-right"
    paginator_theme_prev_icon = "listing-icon-left"
    paginator_prev_text = "Previous page"
    paginator_next_text = "Next page"
    paginator_theme_first_icon = "listing-icon-to-start-alt"
    paginator_theme_last_icon = "listing-icon-to-end-alt"
    paginator_first_text = "First page"
    paginator_last_text = "Last page"
    paginator_in_footer = True
    paginator_has_page_info = True
    filters = Filters(
        Filter(
            "transaction_id",
            filter_key="transaction_id__icontains",
            help_text="Case insensitive",
        ),
    )
    filters.form_layout = (
        "transaction_id;"
    )
    filters.form_buttons = "submit,reset"
    # remove default behaviour when there no row to display (remove the listing
    # and display the template 'empty_listing_template_name' )
    empty_listing_template_name = None
    # instead, display a string into the empty table itself.
    empty_table_msg = "There are no trackers corresponding to your criteria"
    per_page = 16

```

<img width="1087" alt="image" src="https://github.com/elapouya/django-listing/assets/48888439/c361b39f-3b9b-411c-97a5-96329c671af3">

**Comment by elapouya** [comment]
Hello,

What version are you using ?

At first sight, your code seems ok. But, why are you using in your view.py : 
```
context_classes = (
        ToolbarListing,
        Tracker,
    )
```
As you are using instance of listing you do not need to create a listing at template side : should give me the whole template file....

Some ideas : 

* There may be a type with an invisible char so that `tracker` variable is not exactly `tracker` (try cat -etv on your template file)
* instead of putting `tracker` in the context and template, use `listing` 
* try to reduce the equation : make your code as simple as the possible : try to replicate the code from showcase : 
see : showcase/demo/templates/demo/filters1.html and showcase/demo/views.py:296

I made myself a similar test with `tracker` variable, but it works perfectly.
For more help, you have to give me a minimal example so I can reproduce your problem

**Comment by menonsoftware** [comment]
Hello Eric,

This got sorted by using a "Listing" instead of "ListingVariation". Now I'm facing an issue listing the index of records in the table. I can get the index using **rec.get_index()** in a custom Column, however the index resets on every page as I'm using pagination. How can I persist the index counter when moving to another page?

Many thanks for your help!

**Comment by elapouya** [comment]
Could you try : `rec.get_listing().paginator.start_index() + rec.get_index()`

**Comment by menonsoftware** [comment]
> Could you try : `rec.get_listing().paginator.start_index() + rec.get_index()`

Gives me an error "'Paginator' object has no attribute 'start_index'"

**Comment by elapouya** [comment]
Sorry, start_index() is a page method, so try `rec.get_listing().current_page.start_index() + rec.get_index()`

Like previous tip, I did not tested that yet.

**Comment by menonsoftware** [comment]
Works like a charm!

**Comment by elapouya** [comment]
Added LineNumberColumn in v0.0.19

**Comment by menonsoftware** [comment]
Hi Eric,

You may want to return this as a string or so. Else, during an export, using "ExportSelectToolbarItem" it gives an **_AttributeError:
Listing object has no attribute 'current_page'_**. Also, the export shows "-" in the counter rows once exported (if using str).

**Comment by elapouya** [comment]
fixed in 0.0.20
